tests/ecc.rs

```diff
@@ -430,3 +430,141 @@ fn assert_equal_point_works() {
+
+    let n = 1 << 4;
+    let label = b"demo";
+    let pp = PublicParameters::setup(n, rng).expect("failed to create pp");
+
+    pub struct DummyCircuit {
+        point: JubJubAffine,
+        public: JubJubAffine,
+    }
+
+    impl DummyCircuit {
+        pub fn new(point: JubJubAffine, public: JubJubAffine) -> Self {
+            Self { point, public }
+        }
+    }
+
+    impl Default for DummyCircuit {
+        fn default() -> Self {
+            Self {
+                point: dusk_jubjub::GENERATOR,
+                public: dusk_jubjub::GENERATOR,
+            }
+        }
+    }
+
+    impl Circuit for DummyCircuit {
+        fn circuit<C>(&self, composer: &mut C) -> Result<(), Error>
+        where
+            C: Composer,
+        {
+            let w_point = composer.append_point(self.point);
+            composer.assert_equal_public_point(w_point, self.public);
+
+            Ok(())
+        }
+    }
+
+    let (prover, verifier) = Compiler::compile::<DummyCircuit>(&pp, label)
+        .expect("failed to compile circuit");
+
+    // Test default works:
+    // GENERATOR = GENERATOR
+    {
+        let (proof, public_inputs) = prover
+            .prove(rng, &Default::default())
+            .expect("prover shouldn't fail");
+
+        assert_eq!(
+            public_inputs,
+            vec![
+                dusk_jubjub::GENERATOR.get_x(),
+                dusk_jubjub::GENERATOR.get_y()
+            ],
+            "Public input should be the coordinates of the jubjub generator point"
+        );
+
+        verifier
+            .verify(&proof, &public_inputs)
+            .expect("Default circuit verification should pass");
+    }
+
+    // Test sanity:
+    // 42 * GENERATOR = 42 * GENERATOR
+    {
+        let scalar = JubJubScalar::from(42u64);
+        let point = dusk_jubjub::GENERATOR_EXTENDED * &scalar;
+        let public = dusk_jubjub::GENERATOR_EXTENDED * &scalar;
+        let circuit = DummyCircuit::new(point.into(), public.into());
+
+        let (proof, public_inputs) =
+            prover.prove(rng, &circuit).expect("prover shouldn't fail");
+
+        let public_affine: JubJubAffine = public.into();
+        assert_eq!(
+            vec![public_affine.get_x(), public_affine.get_y()],
+            public_inputs,
+            "Public input should be the coordinates of the jubjub generator
+        point multiplied by 42"
+        );
+
+        verifier
+            .verify(&proof, &public_inputs)
+            .expect("Circuit verification with equal points should pass");
+    }
+
+    // Test:
+    // GENERATOR != 42 * GENERATOR
+    {
+        let scalar = JubJubScalar::from(42u64);
+        let point = dusk_jubjub::GENERATOR;
+        let public = dusk_jubjub::GENERATOR_EXTENDED * &scalar;
+        let circuit = DummyCircuit::new(point, public.into());
+
+        prover
+            .prove(rng, &circuit)
+            .expect_err("prover should fail because the points are not equal");
+    }
```

Comment:
that would mean cloning the prover and verifier for each of these functions. i think it's a bit of an overkill

Reply:
naah, it would need two functions per `TestCircuit`, one for testing a satisfied circuit and one for testing an unsatisfied circuit and I also want to have custom failure messages for each test case. I think it wouldn't make things easier to read in the end. Also the `TestCircuit` itself doesn't exist outside of the test function etc... too complicated